kore/src/Kore/Exec.hs

```diff
@@ -281,41 +282,77 @@ exec
         , rewrites
         , equations
         }
-    strategy
+    execMode
     (mkRewritingTerm -> initialTerm) =
         evalSimplifier simplifierx verifiedModule' sortGraph overloadGraph metadataTools equations $ do
             let Initialized{rewriteRules} = rewrites
             finals <-
-                getFinalConfigsOf $ do
-                    initialConfig <-
-                        Pattern.simplify
-                            (Pattern.fromTermLike initialTerm)
-                            >>= Logic.scatter
-                    let updateQueue = \as ->
-                            Strategy.unfoldDepthFirst as
-                                >=> lift
-                                    . Strategy.applyBreadthLimit
-                                        breadthLimit
-                                        dropStrategy
-                        rewriteGroups = groupRewritesByPriority rewriteRules
-                        transit instr config =
-                            Strategy.transitionRule
-                                ( transitionRule rewriteGroups strategy
-                                    & profTransitionRule
-                                    & trackExecDepth
+                do
```

Comment:
This huge block here can be refactored to be made more readable. I think it's nicer see the main logic that is happening at the top, with helper functions having suggestive names. If I want to see the details of the implementation, I can then look for the definitions of the helpers.